core/src/main/scala/dagr/core/tasksystem/EitherTask.scala

```diff
@@ -49,8 +56,9 @@ object EitherTask {
   *
   * @param left the left task.
   * @param right the right task
-  * @param flag indicates which task to perform.
+  * @param choice an expression that returns either Left or Right when invoked
   */
-class EitherTask(private val left: Task, private val right: Task, var flag: EitherTaskFlag.Value) extends Task {
-  override def getTasks: Traversable[Task] = List(if (flag == EitherTaskFlag.LEFT) left else right)
+class EitherTask private (private val left: Task, private val right: Task, private val choice: () => EitherTask.Choice) extends Task {
+  /** Decides which task to return based on `choice` at execution time. */
+  override def getTasks: Traversable[Task] = Seq(if (choice() eq EitherTask.Left) left else right)
```

Comment:
It has to be `choice()` otherwise it will compare the function reference, not the result of calling the function.